src/SqlPersistence/Config/SqlDialect_MsSqlServer.cs

```diff
@@ -24,8 +26,20 @@ internal override void AddCreationScriptParameters(DbCommand command)
 
             internal override void FillParameter(DbParameter parameter, string paramName, object value)
             {
-                parameter.ParameterName = paramName;
-                parameter.Value = value;
+                if (value is ArraySegment<char> charSegment)
```

Comment:
Isn't this required for MySQL and Oracle as well?

Reply:
`.Offset` and `.Size` are SqlParameter specific and there's no common API for them on the `DbParameter` level. 
I know that with `Npgql` for Postgre you can use this as well, but still, it's db driver specific.

Reply:
👍